python/cudf/cudf/core/multiindex.py

```diff
@@ -1575,10 +1575,17 @@ def droplevel(self, level=-1):
 
     @_cudf_nvtx_annotate
     def to_pandas(self, *, nullable: bool = False) -> pd.MultiIndex:
-        result = self.to_frame(
-            index=False, name=list(range(self.nlevels))
-        ).to_pandas(nullable=nullable)
-        return pd.MultiIndex.from_frame(result, names=self.names)
+        return pd.MultiIndex(
+            levels=[
+                level.to_pandas(nullable=nullable) for level in self.levels
+            ],
+            # np.iinfo.min used as missing code, but pandas uses -1
```

Comment:
I'm not sure if I follow what this comment means.

Is the `np.iinfo.min` coming from libcudf? (More generally, where is that value coming from?)

Should this be a "replace value" rather than a "clip" call?

Should we prefer to call something from libcudf instead of cupy?

Reply:
I think it's coming from libcudf indirectly because it's being used in `MultiIndex.__init__`: https://github.com/rapidsai/cudf/blob/e03623ae2ddbc4326201c30f15540ac04d78c0d6/python/cudf/cudf/core/multiindex.py#L192

> Should this be a "replace value" rather than a "clip" call? Should we prefer to call something from libcudf instead of cupy?

Good point. A "replace value" would be more suitable here. I was kinda blindly following how `MultiIndex.codes` generates cupy arrays and working off that